src/app.ts

```diff
@@ -117,44 +106,154 @@ async function runWizard() {
 
   const fileName = await input({
     message:
-      "What's your primary source? \n Drag a text file (or youtube link, experimental) in here, or leave empty/whitespace to open an editor: ",
-    default: wizardState.primarySourceFilename || undefined,
+      "What's your primary source? \n Drag a folder or text file (or youtube link, experimental) in here, or leave empty/whitespace to open an editor: ",
+    default: wizardState.primarySourceFileOrFolderName || undefined,
     validate: async (filename) => {
-      if(filename?.trim()) {
-        if((filename === wizardState.primarySourceFilename || filename === parsePlatformIndependentPath(filename)) && wizardState.loadedPrimarySource) return true;
+      if (filename?.trim()) {
+        if (
+          (filename === wizardState.primarySourceFileOrFolderName ||
+            wizardState.primarySourceFileOrFolderName === parsePlatformIndependentPath(filename)) &&
+          wizardState.loadedPrimarySource
+        )
+          return true; // return true if we've already loaded the file and it's the same
         if (filename.includes("youtube.com")) {
           try {
             const transcript = await YoutubeTranscript.fetchTranscript(filename);
             wizardState.loadedPrimarySource = transcript.map((line) => line.text).join("\n");
-            wizardState.primarySourceFilename = filename;
-          } catch(err) {
+            wizardState.primarySourceFileOrFolderName = filename;
+          } catch (err) {
             return `Looked like a youtube video - Couldn't fetch transcript from ${filename}: ${err}`;
           }
-        } else if(!fs.existsSync(parsePlatformIndependentPath(filename))) {
-          return `File not found - tried to load ${filename}. Try again.`;
         } else {
-          try {
-            const dataFromFile = fs.readFileSync(parsePlatformIndependentPath(filename), "utf-8");
-            wizardState.loadedPrimarySource = dataFromFile;
-            wizardState.primarySourceFilename = parsePlatformIndependentPath(filename);
-          } catch(err) {
-            return `Couldn't read file - tried to load ${filename}. Try again.`;
+          const parsed_filename = parsePlatformIndependentPath(filename);
+          if (!fs.existsSync(parsed_filename)) return `File not found - tried to load ${filename}. Try again.`;
```

Comment:
Maybe should say 'file or folder not found' for correctness

Reply:
done